sematic/examples/testing_pipeline/pipeline.py

```diff
@@ -53,6 +53,36 @@ def add4_nested(a: float, b: float, c: float, d: float) -> float:
     return add2_nested(add2_nested(a, b), add2_nested(c, d))
 
 
+@sematic.func(inline=True, cache=True)
+def add_inline_cached(a: float, b: float) -> float:
+    """
+    Adds two numbers inline, attempting to source the value from the cache.
+    """
+    logger.info("Executing: add_inline_cached(a=%s, b=%s)", a, b)
+    time.sleep(5)
+    return a + b
+
+
+@sematic.func(inline=False, cache=True)
+def add2_nested_cached(a: float, b: float) -> float:
+    """
+    Adds two numbers using a nested structure, attempting to source the value from the
+    cache.
+    """
+    logger.info("Executing: add2_nested_cached(a=%s, b=%s)", a, b)
+    return add_inline_cached(a, b)
+
+
+@sematic.func(inline=False, cache=True)
+def add4_nested_cached(a: float, b: float, c: float, d: float) -> float:
+    """
+    Adds four numbers using a nested structure, attempting to source the value from the
+    cache.
+    """
+    logger.info("Executing: add4_nested_cached(a=%s, b=%s, c=%s, d=%s)", a, b, c, d)
+    return add2_nested_cached(add2_nested_cached(a, b), add2_nested_cached(c, d))
```

Comment:
I don't think you need to create separate functions for `cache=True`. You can use `.set(cache=cache)` on the futures.